File: Code-Review-Mcp/test-fixtures/untested-api/angular/discriminator-a.ts
// Discriminating fixture (per-class scoping must be falsifiable):
// `compute` is NEVER referenced from THIS class's spec, but the same word
// `compute` appears in DiscriminatorB's spec. With the real import gate the
// member check is scoped to specs that import DiscriminatorA only, so
// `DiscriminatorA.compute` MUST be reported as "no_reference_found". A rollback
// to global (cross-class) member matching would wrongly treat it as referenced
// (false negative) and turn the harness red.
export class DiscriminatorA {
  compute(): number {
    return 1;
  }

  // Referenced from discriminator-a.spec.ts → keeps the class "tested".
  keep(): number {
    return 0;
  }
}

File: Mcp-Servers/Codebase.Analyzer/test-fixtures/untested-api/dotnet/DiscriminatorA.cs
namespace SampleApp;

// Discriminating fixture (per-class scoping must be falsifiable):
// `Compute` is NEVER referenced from DiscriminatorATests, but the same word
// `Compute` appears in DiscriminatorBTests. Because the member check is scoped to
// the test files associated with THIS class only, `DiscriminatorA.Compute` MUST
// be reported as "no_reference_found". A rollback to global (cross-class) member
// matching would wrongly treat it as referenced (false negative) → harness red.
//
// The two `Compute` overloads also exercise member dedup: a single
// `DiscriminatorA.Compute` finding must be emitted, not one per overload.
public class DiscriminatorA
{
    public int Compute() => 1;
    public int Compute(int seed) => seed;

    // Referenced from DiscriminatorATests → keeps the class "tested".
    public int Keep() => 0;
}
